core/src/main/java/com/linecorp/armeria/internal/common/util/SelfSignedCertificate.java

```diff
@@ -350,36 +380,45 @@ public void delete() {
     }
 
     private static String[] generate(String fqdn, KeyPair keypair, Random random,
-                                     Date notBefore, Date notAfter, String algorithm) throws Exception {
+                                     Date notBefore, Date notAfter, String algorithm,
+                                     List<SelfSingedCertificateNameType> nameTypes) throws Exception {
 
         final PrivateKey key = keypair.getPrivate();
 
         // Prepare the information required for generating an X.509 certificate.
-        final X500Name owner = new X500Name("CN=" + fqdn);
+        final X500Name owner;
+        if (nameTypes.contains(SelfSingedCertificateNameType.COMMON_NAME)) {
+            owner = new X500Name("CN=" + fqdn);
+        } else {
+            // Add a dummy OU as an empty DN is not allowed.
+            owner = new X500Name("OU=SelfSigned");
+        }
         final X509v3CertificateBuilder builder = new JcaX509v3CertificateBuilder(
                 owner, new BigInteger(64, random), notBefore, notAfter, owner, keypair.getPublic());
 
-        final List<GeneralName> names = new ArrayList<>();
-        // Add fqdn as Subject Alternative Name too for clients that don't use CN when validating such as
-        // OkHttp.
-        if (NetUtil.isValidIpV4Address(fqdn) || NetUtil.isValidIpV6Address(fqdn)) {
-            names.add(new GeneralName(iPAddress, fqdn));
-        } else {
-            names.add(new GeneralName(dNSName, fqdn));
-        }
+        if (nameTypes.contains(SelfSingedCertificateNameType.SUBJECT_ALTERNATIVE_NAME)) {
```

Comment:
Because using common name is discouraged, can't we remove this `SelfSingedCertificateNameType` and always use san?

Reply:
This was added for testing purposes, especially to allow testing of CN and SAN fields using SelfSignedCertificate.